packages/connection-form/src/components/connection-form.tsx

```diff
@@ -57,54 +51,77 @@ const descriptionStyles = css({
 });
 
 const formStyles = css({
-  display: 'contents',
+  display: 'grid',
+  gridTemplateAreas: `
+    "header"
+    "content"
+    "footer"
+  `,
+  gridTemplateRows: 'max-content 1fr max-content',
+
+  minHeight: 0,
+  height: '100%',
 });
 
-const formContainerStyles = css({
-  padding: spacing[4],
-  paddingRight: spacing[3], // this is to leave space for the scrollbarGutter
-  paddingBottom: spacing[1],
-  overflowY: 'auto',
+const formHeaderStyles = css({
+  gridArea: 'header',
+
+  paddingTop: spacing[600],
+  paddingLeft: spacing[600],
+  paddingRight: spacing[600],
+});
+
+const formHeaderContentStyles = css({
   position: 'relative',
 });
 
-const formContentStyles = css({
+const formContentContainerStyles = css({
+  gridArea: 'content',
+
   display: 'flex',
-  columnGap: spacing[3],
-  maxHeight: '720px',
-  overflow: 'auto',
+  gap: spacing[400],
+
+  paddingLeft: spacing[600],
+  paddingRight: spacing[600],
+
+  overflowY: 'auto',
   scrollbarGutter: 'stable',
 });
 
-const formContentLegacyStyles = css({
-  height: 'auto',
+const formContentStyles = css({
+  minWidth: 0,
+  flex: '1 1 auto',
+
+  // To prevent input outline being cut by the overflow
+  paddingLeft: spacing[100],
 });
 
-const formSettingsStyles = css({
-  width: '100%',
-  paddingLeft: spacing[1],
-  paddingRight: spacing[1],
+const formHelpConatinerStyles = css({
+  position: 'sticky',
```

Comment:
Don't know why I didn't think of or realise you can do position sticky inside here. I mean obviously when I think about it, but for some reason in my mind it was a relative to the screen thing.